Sources/Workspace/Workspace.swift

```diff
@@ -296,11 +305,17 @@ public class Workspace {
             delegate: delegate.map(WorkspaceRepositoryManagerDelegate.init(workspaceDelegate:)),
             cachePath: sharedRepositoriesCacheEnabled ? location.sharedRepositoriesCacheDirectory : .none
         )
+        let registryManager = customRegistryManager ?? registries.map { configuration in
+            RegistryManager(configuration: configuration,
+                            identityResolver: identityResolver,
+                            authorizationProvider: authorizationProvider?.httpAuthorizationHeader(for:),
+                            diagnosticEngine: nil)
```

Comment:
its unlikely to have a diagnosticEngine at this point since workspace is long lived. probably worth removing from the ctor and moving the public methods that need it

Reply:
Thanks for calling this out. I have a note to myself to move `diagnosticEngine` to parameters in the instance methods themselves. Working on that now.

Reply:
Fixed by 9d1c129b